### PR TITLE
crimson: improve dump_ops_in_flight

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
This was really handy for debugging https://tracker.ceph.com/issues/57740

File: src/crimson/osd/osd_operations/peering_event.h [hunk 1]
```diff
@@ -133,11 +133,6 @@ class RemotePeeringEvent : public PeeringEvent<RemotePeeringEvent> {
     conn(conn)
   {}
 
-#if 0
```
Comment (rzarzynski) [inline src/crimson/osd/osd_operations/peering_event.h:136]: Thanks!

File: src/crimson/admin/osd_admin.cc [hunk 2]
```diff
@@ -452,27 +453,32 @@ template std::unique_ptr<AdminSocketHook> make_asok_hook<InjectMDataErrorHook>(
...
 				      std::string_view format,
 				      ceph::bufferlist&& input) const final
   {
-    logger().warn("{}", __func__);
-    unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+    unique_ptr<Formatter> fref{
+      Formatter::create(format, "json-pretty", "json-pretty")};
+    auto *f = fref.get();
     f->open_object_section("ops_in_flight");
-    op_registry.dump_client_requests(f.get());
```
Comment (rzarzynski) [inline src/crimson/admin/osd_admin.cc:457]: Not sure if `dump_client_requests()` is used anywhere after that.

Comment (rzarzynski) [reply src/crimson/admin/osd_admin.cc:457]:
```
$ grep -r dump_client_requests crimson/
crimson/osd/osd_operation.cc:size_t OSDOperationRegistry::dump_client_requests(ceph::Formatter* f) const
crimson/osd/osd_operation.h:  size_t dump_client_requests(ceph::Formatter* f) const;
crimson/admin/osd_admin.cc:    op_registry.dump_client_requests(f.get());
```

**Review by rzarzynski — APPROVED** [review]

**Pull request merged**